gammapy/scripts/downloadclasses.py

```diff
@@ -232,11 +215,19 @@ def run(self):
                 md5 = self.listfiles[rec]["hashmd5"]
             ifolder = Path(path).parent
             ifolder.mkdir(parents=True, exist_ok=True)
-            ftuple = (url, str(path), md5)
-            pool.apply_async(get_file, args=(ftuple,), callback=self.progressbar)
-        pool.close()
-        pool.join()
-        pool.close()
+            retrieve = True
+            if md5 and path.exists():
+                md5local = hashlib.md5(path.read_bytes()).hexdigest()
```

Comment:
I was hoping parfive would allow us to get rid of urllib completely, and also code that checks if files already have been downloaded. Can you try to go that way, to use it for everything?
If some feature is missing (such as e.g. checksum check to avoid downloading multiple times), you could file a feature request with parfive?